lib/normalizer.c

```diff
@@ -909,15 +902,53 @@ grn_nfkc_normalize_unify_alphabet_diacritical_mark(
        * U+1EB3 LATIN SMALL LETTER A WITH BREVE AND HOOK ABOVE
        * U+1EB5 LATIN SMALL LETTER A WITH BREVE AND TILDE
        * U+1EB7 LATIN SMALL LETTER A WITH BREVE AND DOT BELOW
-       * Uppercase counterparts(U+1EA2, U+1EA4, U+1EA6, U+1EA8, U+1EAA,
+       * Uppercase counterparts (U+1EA2, U+1EA4, U+1EA6, U+1EA8, U+1EAA,
        * U+1EAC, U+1EAE, U+1EB0, U+1EB2, U+1EB4, and U+1EB6) are covered by the
        * following condition but they are never appeared here. Because NFKC
        * normalization converts them to their lowercase equivalents.
        */
       (utf8_char[0] == 0xE1 && utf8_char[1] == 0xBA &&
        (0xA1 <= utf8_char[2] && utf8_char[2] <= 0xB7))) {
+    return true;
+  }
+  return false;
+}
+
+grn_inline static bool
+grn_nfkc_normalize_unify_diacritical_mark_is_b(const unsigned char *utf8_char)
+{
+  /*
+   * Latin Extended Additional
+   * U+1E03 LATIN SMALL LETTER B WITH DOT ABOVE
+   * U+1E05 LATIN SMALL LETTER B WITH DOT BELOW
+   * U+1E07 LATIN SMALL LETTER B WITH LINE BELOW
+   * Uppercase counterparts (U+1E04, U+1E06) are covered by the
+   * following condition but they are never appeared here. Because NFKC
+   * normalization converts them to their lowercase equivalents.
+   */
+  if (utf8_char[0] == 0xE1 && utf8_char[1] == 0xB8 &&
+      (0x83 <= utf8_char[2] && utf8_char[2] <= 0x87)) {
+    return true;
+  }
+  return false;
```

Comment:
`if (...) {return true;} else {return false;}`ってたまに見るんですけど、こういうのは`return ...;`にしてください。

```suggestion
  return (utf8_char[0] == 0xE1 && utf8_char[1] == 0xB8 &&
      (0x83 <= utf8_char[2] && utf8_char[2] <= 0x87));
```

Reply:
fix: [7f62d67](https://github.com/groonga/groonga/pull/1891/commits/7f62d6707b27b2146407b13a3d221a324bdd169a) 真偽値の判定以外に何も行っていないので、そのまま返すように変更しました 🙏🏾